tests/test_types_typeddict.py

```diff
@@ -905,3 +906,16 @@ class MySubTypedDict(MyMiddleTypedDict):
 
     validated_data = TypeAdapter(MySubTypedDict).validate_python({'x': 'ABC', 'y': 'DEF', 'z': 'GHI'})
     assert validated_data == {'x': 'abc', 'y': 'def', 'z': 'ghi'}
+
+
+def test_typeddict_mro():
+    class A(TypedDict):
+        x = 1
+
+    class B(A):
+        x = 2
+
+    class C(B):
+        pass
+
+    assert get_attribute_from_bases(C, 'x') == 2
```

Comment:
Currently on the `main` branch, this returns `1`

Reply:
Oh wow I’m very glad you caught this 🙈